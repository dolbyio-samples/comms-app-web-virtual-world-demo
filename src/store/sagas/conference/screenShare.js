import { call } from "redux-saga/effects";

import {
  startScreenShare as start,
  stopScreenShare as stop,
} from "../../services/screenShare";
import { displayMessage } from "./message";

export function* startScreenShare(action) {
  const { error } = yield call(start, { audio: true });
  if (error) {
    yield call(displayMessage, {
      type: "error",
      message: "could not start screen sharing",
    });
  }
}

export function* stopScreenShare() {
  const { error } = yield call(stop);
  if (error) {
    yield call(displayMessage, {
      type: "error",
      message: "could not stop screen sharing",
    });
  }
}
