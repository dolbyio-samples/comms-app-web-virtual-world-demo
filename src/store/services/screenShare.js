import VoxeetSDK from "@voxeet/voxeet-web-sdk";

export const startScreenShare = (option = { audio: true }) => {
  return VoxeetSDK.conference
    .startScreenShare(option)
    .then((response) => ({ response }))
    .catch((error) => ({ error }));
};

export const stopScreenShare = () => {
  return VoxeetSDK.conference
    .stopScreenShare({ audio: true })
    .then((response) => ({ response }))
    .catch((error) => ({ error }));
};
