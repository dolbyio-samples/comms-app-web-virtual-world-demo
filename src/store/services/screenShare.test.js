import { describe, it, expect, vi } from "vitest";
import { startScreenShare, stopScreenShare } from "./screenShare";

const mockStartScreenShare = vi.fn();
const mockStopScreenShare = vi.fn();

vi.mock("@voxeet/voxeet-web-sdk", () => ({
  default: {
    get conference() {
      return {
        startScreenShare: mockStartScreenShare,
        stopScreenShare: mockStopScreenShare,
      };
    },
  },
}));

describe("screen share services test suite", () => {
  it("should return a resolved promise", async () => {
    // arrange
    mockStartScreenShare.mockResolvedValue({ response: {} });
    const option = { audio: true };

    // act
    const { response, error } = await startScreenShare(option);

    // assert
    expect(mockStartScreenShare).toHaveBeenCalled();
    expect(mockStartScreenShare).toHaveBeenCalledWith(option);
    expect(response).toMatchSnapshot();
    expect(error).toBe(undefined);
  });

  it("should return a resolved promise", async () => {
    // arrange
    mockStopScreenShare.mockResolvedValue(true);

    // act
    const { response, error } = await stopScreenShare();

    // assert
    expect(mockStopScreenShare).toHaveBeenCalled();
    expect(response).toMatchSnapshot();
    expect(error).toBe(undefined);
  });

  it("should return a resolved promise containing an error", async () => {
    // arrange
    mockStartScreenShare.mockRejectedValueOnce(new Error("error"));

    // act
    const { response, error } = await startScreenShare();

    // assert
    expect(mockStartScreenShare).toHaveBeenCalled();
    expect(response).toBe(undefined);
    expect(error.message).toBe("error");
  });

  it("should return a resolved promise containing an error", async () => {
    // arrange
    mockStopScreenShare.mockRejectedValueOnce(new Error("error"));

    // act
    const { response, error } = await stopScreenShare();

    // assert
    expect(mockStopScreenShare).toHaveBeenCalled();
    expect(error.message).toBe("error");
    expect(response).toBe(undefined);
  });
});
