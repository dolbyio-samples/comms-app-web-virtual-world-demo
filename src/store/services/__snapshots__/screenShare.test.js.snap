// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`screen share services test suite > should return a resolved promise 1`] = `
{
  "response": {},
}
`;

exports[`screen share services test suite > should return a resolved promise 2`] = `true`;
